src/ripple/app/misc/impl/Transaction.cpp

```diff
@@ -132,6 +133,23 @@ Transaction::pointer Transaction::load(uint256 const& id, Application& app)
         soci::blob sociRawTxnBlob (*db);
         soci::indicator rti;
 
+        if(searchedAll)
+        {
```

Comment:
I would only do this calculation if we don't find the txn. (See other comment about changing the function signature to make it clear this function either returns the txn or if the range was searched).

Reply:
Done